### PR TITLE
Add order-level telephone field to core

### PR DESCRIPTION
**Pull request by raphaelm**
This basically moves @felixrindt's plugin into core.

TODO:

- [x] Add to export(s)
- [x] Add to shredders
- [x] Migrate data from plugin
- [x] Add tests
- [x] Expose field in API
- [x] Show on presale order details page
- [x] Show in backend
- [x] Allow to modify in backend and frontend
- [x] Add PDF variable

after merge:

- [ ] add to corona shredder

This also changes the layout of a settings dialog:

### CODE REVIEW

File: src/pretix/base/services/orders.py [hunk 1]
```diff
@@ -772,10 +772,12 @@ def _create_order(event: Event, email: str, positions: List[CartPosition], now_d
         fees, pf, gift_card_values = _get_fees(positions, payment_provider, address, meta_info, event, checked_gift_cards)
         total = pending_sum = sum([c.price for c in positions]) + sum([c.value for c in fees])
 
+        print(meta_info)
```
Comment (pc-coholic) [inline src/pretix/base/services/orders.py:775]:
```suggestion
```

File: src/pretix/base/settings.py [hunk 2]
```diff
@@ -1845,6 +1864,20 @@ def unserialize(cls, s):
                         "why you need information from them.")
         )
     },
+    'checkout_phone_helptext': {
+        'default': LazyI18nString.from_gettext(gettext_noop(
+            'Make sure to enter a valid email address. We will send you an order '
+            'confirmation including a link that you need to access your order later.'
```
Comment (wiffbi) [inline src/pretix/base/settings.py:1870]: Update helptext for phone-number.

File: src/pretix/base/settings.py [hunk 3]
```diff
@@ -1845,6 +1864,20 @@ def unserialize(cls, s):
...
+        )),
+        'type': LazyI18nString,
+        'serializer_class': I18nField,
+        'form_class': I18nFormField,
+        'form_kwargs': dict(
+            label=_("Help text of the email field"),
```
Comment (wiffbi) [inline src/pretix/base/settings.py:1876]: Update from „email field“ to „phone number field“

File: src/pretix/presale/forms/checkout.py [hunk 4]
```diff
@@ -31,6 +38,35 @@ def __init__(self, *args, **kwargs):
                 help_text=_('Please enter the same email address again to make sure you typed it correctly.'),
             )
 
+        if self.event.settings.order_phone_asked:
+            babel_locale = 'en'
+            # Babel, and therefore django-phonenumberfield, do not support our custom locales such das de_Informal
```
Comment (wiffbi) [inline src/pretix/presale/forms/checkout.py:43]:
```suggestion
            # Babel, and therefore django-phonenumberfield, do not support our custom locales such as de_Informal
```
Nur ein Mini-Typo. Sorry, will auf keinen Fall pedantisch sein.

File: src/pretix/base/settings.py (hunk 3 above)
Comment (wiffbi) [reply src/pretix/base/settings.py:1876]:
```suggestion
            label=_("Help text of the phone number field"),
```

File: src/pretix/presale/forms/checkout.py (hunk 4 above)
Comment (raphaelm) [reply src/pretix/presale/forms/checkout.py:43]: Keine Sorge :)

**Pull request merged**